storm-core/src/jvm/org/apache/storm/daemon/supervisor/BasicContainer.java

```diff
@@ -0,0 +1,628 @@
+import org.apache.storm.utils.LocalState;
+import org.apache.storm.utils.Utils;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.google.common.base.Joiner;
+import com.google.common.collect.Lists;
+
+/**
+ * A container that runs processes on the local box.
+ */
+public class BasicContainer extends Container {
+    private static final Logger LOG = LoggerFactory.getLogger(BasicContainer.class);
+    private static final FilenameFilter jarFilter = new FilenameFilter() {
+        @Override
+        public boolean accept(File dir, String name) {
+            return name.endsWith(".jar");
+        }
+    };
+    private static final Joiner CPJ = 
+            Joiner.on(Utils.CLASS_PATH_SEPARATOR).skipNulls();
+    
+    protected final LocalState _localState;
+    protected final String _profileCmd;
+    protected volatile boolean _exitedEarly = false;
+
+    private class ProcessExitCallback implements ExitCodeCallback {
+        private final String _logPrefix;
+
+        public ProcessExitCallback(String logPrefix) {
+            _logPrefix = logPrefix;
+        }
+
+        @Override
+        public void call(int exitCode) {
+            LOG.info("{} exited with code: {}", _logPrefix, exitCode);
+            _exitedEarly = true;
+        }
+    }
+
+    //For testing purposes
+    public BasicContainer(AdvancedFSOps ops, int port, LocalAssignment assignment,
+            Map<String, Object> conf, Map<String, Object> topoConf, String supervisorId, 
+            ResourceIsolationInterface resourceIsolationManager, LocalState localState,
+            String profileCmd) throws IOException {
+        super(ops, port, assignment, conf, topoConf, supervisorId, resourceIsolationManager);
+        _localState = localState;
+        _profileCmd = profileCmd;
+    }
+    
+    public BasicContainer(int port, LocalAssignment assignment, Map<String, Object> conf, String supervisorId,
+            LocalState localState, ResourceIsolationInterface resourceIsolationManager, boolean recover)
+            throws IOException {
+        super(port, assignment, conf, supervisorId, resourceIsolationManager);
+        _localState = localState;
+
+        if (recover) {
+            synchronized (localState) {
+                String wid = null;
+                Map<String, Integer> workerToPort = localState.getApprovedWorkers();
+                for (Map.Entry<String, Integer> entry : workerToPort.entrySet()) {
+                    if (port == entry.getValue().intValue()) {
+                        wid = entry.getKey();
+                    }
+                }
+                if (wid == null) {
+                    throw new ContainerRecoveryException("Could not find worker id for " + port + " " + assignment);
+                }
+                _workerId = wid;
+            }
+        } else {
+            createNewWorkerId();
+        }
+
+        String stormHome = System.getProperty("storm.home");
```

Comment:
Minor: `System.getProperty("storm.home")` might be better to place to Utils or so since there're some other places which uses this. I'm also OK to keep this as it is and address this from other issue.